### PR TITLE
multi: take over multiple small stale PRs

### PR DESCRIPTION
**Pull request by guggero**
This PR contains multiple independent changes collected from old, stale PRs.
Replaces:
 - #3461
 - #3283
 - #2649
 - #2599

Original PR authors are attributed where possible.

Because 4 of the 5 replaced PRs somehow involve Docker and development builds, I added a commit of my own to finally clear up this discussion (by tacking my own personal opinion on top of it).

### CODE REVIEW

File: peer/brontide.go [hunk 1]
```diff
@@ -1836,8 +1838,6 @@ out:
 		}
 	}
 
-	p.wg.Done()
```
Comment (carlaKC) [inline peer/brontide.go:1839]: Slight behaviour change here:
- Previously: `wg.Done` then disconnect
- Now: disconnect then `wg.Done`

Seems like this new ordering is the right way, but perhaps just check that nothing strange happens with the change?

File: docs/DOCKER.md [hunk 2]
```diff
@@ -72,23 +72,23 @@ To test the Docker production image locally, run the following from
...
 ```
-$ docker build . --build-arg checkout=$(git describe --tags `git rev-list --tags --max-count=1`) -t lnd:latest-tag
+$ docker build . --build-arg checkout=$(git describe --tags `git rev-list --tags --max-count=1`) -t myrepository/lnd:latest-tag
 ```
 
 Once the image has been built and tagged locally, start the container:
 
 ```
-docker run --name=lnd-testnet -it lnd:1.0 --bitcoin.active --bitcoin.testnet --bitcoin.node=neutrino --neutrino.connect=faucet.lightning.community
+docker run --name=lnd-testnet -it myrepository/lnd:1.0 --bitcoin.active --bitcoin.testnet --bitcoin.node=neutrino --neutrino.connect=faucet.lightning.community
```
Comment (carlaKC) [inline docs/DOCKER.md:109]: Should we change the `1.0` here?

**Review by carlaKC — APPROVED** [review]
Great cleanup!

File: docs/DOCKER.md [hunk 3]
```diff
@@ -1,9 +1,25 @@
 # Docker Instructions
 
+There are two flavors of Dockerfiles available:
+ - `Dockerfile`: Used for production builds. Checks out the source code from
+   GitHub during build. The build argument `--build-arg checkout=v0.x.x-beta`
+   can be used to specify what git tag or commit to check out before building.
+ - `Dockerfile.dev` Used for development or testing builds. Uses the local code
```
Comment (bhandras) [inline docs/DOCKER.md:7]: nit: `dev.Dockerfile`

**Review by bhandras — APPROVED** [review]
LGTM

File: docs/DOCKER.md (hunk 3 above)
Comment (guggero) [reply docs/DOCKER.md:7]: Oh. Thanks, fixed!

File: peer/brontide.go (hunk 1 above)
Comment (guggero) [reply peer/brontide.go:1839]: Ah, good catch! I looked at the initial commit message that added the `p.wg.Done()` there and it explicitly mentioned a deadlock. So I reverted this change and replaced it with that comment instead.

**Pull request merged**